fem/intrules.cpp

```diff
@@ -1648,6 +1656,30 @@ IntegrationRule *IntegrationRules::TetrahedronIntegrationRule(int Order)
    }
 }
 
+// Integration rules for reference pyramid
```

Comment:
hp3d (Leszek Demkowicz's code) also uses Duffy quadrature. This may be fine for higher-order as well

Reply:
This is still an open question in my mind. The integration rules in this PR seem to be sufficient for these first order basis functions. I plan to revisit this when implementing and testing the arbitrary order basis functions. I'd be interested in discussing this with you further when I get to that stage.